src/ViewToggleButton/ViewToggleButton.test.js

```diff
@@ -0,0 +1,50 @@
+import { render, screen } from "@testing-library/react";
+
+import React from "react";
+import ViewToggleButton from ".";
+import userEvent from "@testing-library/user-event";
+
+/**
+ * Test wrapper for ViewToggleButton
+ *
+ * Provides state for value to avoid errors changing from uncontrolled to controlled.
+ */
+const ViewToggleButtonWithState = ({
+  onChange,
+  value: valueIn = "",
+  ...rest
+}) => {
+  const [value, setValue] = React.useState(valueIn);
+  const handleChange = (event, value) => {
+    setValue(value);
+    onChange && onChange(event, value);
+  };
+  return <ViewToggleButton {...rest} onChange={handleChange} value={value} />;
+};
+
+/**
+ * Tests
+ */
+describe("Select", () => {
+  test("can select card view", () => {
+    render(<ViewToggleButtonWithState value="card" />);
+    const cardButton = screen.getByTestId("cardButton");
+    const tableButton = screen.getByTestId("tableButton");
+    expect(cardButton.getAttribute("aria-pressed")).toBe("true");
+    expect(tableButton.getAttribute("aria-pressed")).toBe("false");
+  });
+  test("can select table view", () => {
+    render(<ViewToggleButtonWithState value="table" />);
+    const cardButton = screen.getByTestId("cardButton");
+    const tableButton = screen.getByTestId("tableButton");
+    expect(cardButton.getAttribute("aria-pressed")).toBe("false");
+    expect(tableButton.getAttribute("aria-pressed")).toBe("true");
+  });
+  test("test onchange callback is called on click of an option", async () => {
+    const onChange = jest.fn();
+    render(<ViewToggleButtonWithState onChange={onChange} value="card" />);
```

Comment:
I think what you've done here is written a test that proves that this ViewToggleButton component does not work as expected.

I should never be able to unselect all options, but what you're doing here is unselecting card, leaving nothing selected.

I've just tested this in storybook to confirm:

<img width="426" alt="image" src="https://user-images.githubusercontent.com/27866636/229735817-9e6e347e-e2c8-4917-9956-09a3c5aba94a.png">

I suggest your ViewToggleButton only triggers the onChange callback when you have a value to set, and it is not equal to the current selection. See this section in the MUI docs for some detail: https://mui.com/material-ui/react-toggle-button/#enforce-value-set

Reply:
I now expect a test that checks that the callback is not triggered when you click the currently selected button, as well as another test that checks that it is triggered when I click the other view button.

Reply:
Also, you can now see the danger of just testing that a callback was called, rather than checking what args it was called with.